### PR TITLE
Fix help text in goto-harness

### PR DESCRIPTION
**Pull request by thk123**
While debugging a separate error, noticed an error in the help, and an error message that could be clearer.

The (relevant section of the) updated help now reads:

```
 goto-harness <in> <out> --harness-function-name <name> --harness-type <harness-type> [harness options]

<in>                       goto binaries to read from
<out>                      C file to write the harness to
```

The error message produced is now:

```
Invalid User Input
Option: --associated-array-size
Reason: could not find parameter to associate the array size of array
"arr" (Expected parameter: "siz" on function "test2" in file example.c
line 9)
```

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [na] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [na] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/goto-harness/goto_harness_parse_options.cpp [hunk 1]
```diff
@@ -174,7 +174,8 @@ void goto_harness_parse_optionst::help()
     << " goto-harness <in> <out> --harness-function-name <name> --harness-type "
        "<harness-type> [harness options]\n"
     << "\n"
-    << "<in> <out>                 goto binaries to read from / write to\n"
+    << "<in>                       goto binaries to read from / write to\n"
+    << "<out>                      C file to write the harness to\n"
```
Comment (hannes-steffenhagen-diffblue) [inline src/goto-harness/goto_harness_parse_options.cpp:178]: Yeah this should have been here to begin with

File: src/goto-harness/function_call_harness_generator.cpp [hunk 2]
```diff
@@ -508,7 +508,11 @@ function_call_harness_generatort::implt::declare_arguments(
         associated_array_size_argument == parameter_name_to_argument_name.end())
       {
         throw invalid_command_line_argument_exceptiont{
-          "associated array size is not there",
+          "could not find parameter to associate the array size of array \"" +
+            id2string(parameter_name) + "\" (Expected parameter: \"" +
+            id2string(associated_array_size_parameter) + "\" on function \"" +
+            id2string(function_to_call.display_name()) + "\" in " +
+            function_to_call.location.as_string() + ")",
```
Comment (hannes-steffenhagen-diffblue) [inline src/goto-harness/function_call_harness_generator.cpp:515]: Yeah this is probably for the better. For a second I was wondering whether the redundancy of repeating the function name would be a problem, but honestly it shouldn’t hurt and might help when running this from a script.

**Review by hannes-steffenhagen-diffblue — APPROVED** [review]

File: src/goto-harness/goto_harness_parse_options.cpp [hunk 3]
```diff
@@ -174,7 +174,8 @@ void goto_harness_parse_optionst::help()
     << " goto-harness <in> <out> --harness-function-name <name> --harness-type "
        "<harness-type> [harness options]\n"
     << "\n"
-    << "<in> <out>                 goto binaries to read from / write to\n"
+    << "<in>                       goto binaries to read from / write to\n"
```
Comment (danpoe) [inline src/goto-harness/goto_harness_parse_options.cpp]: Should this say "goto binary to read from"?

**Review by danpoe — APPROVED** [review]

File: src/goto-harness/function_call_harness_generator.cpp (hunk 2 above)
Comment (thk123) [reply src/goto-harness/function_call_harness_generator.cpp:515]: Where is the function repeated? (for convenience I attached the new error message to the PR description)

File: src/goto-harness/goto_harness_parse_options.cpp (hunk 3 above)
Comment (thk123) [reply src/goto-harness/goto_harness_parse_options.cpp]: :facepalm:

**Pull request merged**